sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -771,6 +771,17 @@ object LimitPushDown extends Rule[LogicalPlan] {
       Limit(le, Project(a.aggregateExpressions, LocalLimit(le, a.child)))
     case Limit(le @ IntegerLiteral(1), p @ Project(_, a: Aggregate)) if a.groupOnly =>
       Limit(le, p.copy(child = Project(a.aggregateExpressions, LocalLimit(le, a.child))))
+    // Push down limit 1 though Union and Aggregate
+    case Limit(le @ IntegerLiteral(1), u: Union) =>
+      val newUnionChildren = u.children.map {
+        case a: Aggregate if a.groupOnly =>
```

Comment:
I doubt that the output will be changed.
`limit 1 after shuffle` is not the same as `limit 1 before shuffle`

Reply:
We have supported this before:
https://github.com/apache/spark/blob/ddb36f9e0151d3bff8b9c8a2e3bb14a6f6ece218/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala#L769-L773